web/backend/plugins/metrics.ts

```diff
@@ -1,10 +1,41 @@
 import { FastifyInstance } from 'fastify'
 import { getMetrics } from '../utils/metrics'
+import { join } from 'path'
+import { writeFile } from 'fs/promises'
+import { Mode } from '../schemas'
+import { getPidToLoad, getSelectableRuntimes } from '../utils/runtimes'
+import { RuntimeApiClient } from '@platformatic/control'
+import { MS_WAITING } from '../utils/constants'
 
 export default async function (fastify: FastifyInstance) {
+  const isValidMode = (mode: Mode) => fastify.loaded.mode === mode && prevMode !== mode
+
+  const api = new RuntimeApiClient()
+  let prevMode: Mode | undefined
+
   fastify.decorate('mappedMetrics', {})
 
-  fastify.decorate('metricsInterval', setInterval(() => getMetrics(fastify), 1000))
+  fastify.decorate('loaded', {
+    runtimes: [],
+    services: { entrypoint: '', production: false, services: [] },
+    path: join(__dirname, '..', '..', '..', 'frontend', 'src', 'loaded.json')
+  })
+
+  fastify.decorate('metricsInterval', setInterval(async () => {
+    getMetrics(fastify)
+
+    if (isValidMode('start')) {
+      fastify.mappedMetrics = {}
+    }
+
+    if (isValidMode('stop')) {
+      const runtimes = getSelectableRuntimes(await api.getRuntimes(), false)
+      const services = await api.getRuntimeServices(getPidToLoad(runtimes))
+      await writeFile(fastify.loaded.path, JSON.stringify({ runtimes, services, metrics: fastify.mappedMetrics[getPidToLoad(runtimes)] }))
+    }
```

Comment:
This looks like to not work in practice, we can't really execute this on an interval, but in the `/record` handler